server/service.go

```diff
@@ -408,6 +408,9 @@ func (m *BoostService) handleGetHeader(w http.ResponseWriter, req *http.Request)
 			// Compare the bid with already known top bid (if any)
 			if result.response.Data != nil {
 				valueDiff := responsePayload.Data.Message.Value.Cmp(&result.response.Data.Message.Value)
+
+				log.Infof("Relay %s offered bid with value %s", relay.String(), responsePayload.Data.Message.Value.String())
```

Comment:
relay and value are already added as fields to `log` at this point, and to reduce verbosity I'd change this to `log.Debug`. So the function would look like `log.Debug("bid received")

Reply:
you probably also want to log this above the `if` in 409, because currently it would only log if it's not the first received bid

Reply:
> relay and value are already added as fields to `log` at this point, and to reduce verbosity I'd change this to `log.Debug`. So the function would look like `log.Debug("bid received")`

Ah, so that's what the `log.withField` stuff is for. Noted, and indeed will just change it to a simple "bid received" then :)

 > you probably also want to log this above the if in 409, because currently it would only log if it's not the first received bid
And that's why I shouldn't do late-night programming, well-caught!

I do wonder about the `log.Debug("received a good bid")` at the end now though. Something along the lines of "using bid as response" or "selecting bid" would be more descriptive and signify the difference with the new log message. I now picked the latter, because it's the shortest.